test/integration/integration.test.ts

```diff
@@ -109,493 +65,242 @@ describe('Integration tests', (): void => {
       )
-
-      const response = await c9.adminClient.createOutgoingPayment({
-        walletAddressId: gfranklinWalletAddressId,
-        quoteId: quote.id
-      })
-
-      expect(response.code).toBe('200')
-      assert(response.payment)
-
-      outgoingPayment = response.payment
-
-      await pollCondition(
-        () => {
-          return (
-            handleWebhookEventSpy.mock.calls.some(
-              (call) =>
-                call[0]?.type === WebhookEventType.OutgoingPaymentCreated
-            ) &&
-            handleWebhookEventSpy.mock.calls.some(
-              (call) =>
-                call[0]?.type === WebhookEventType.OutgoingPaymentCompleted
-            )
-          )
+      assert(senderWalletAddress)
+      const senderAssetCode = senderWalletAddress.assetCode
+      const senderWalletAddressId = senderWalletAddress.walletAddressID
+      const value = '500'
+      const createReceiverInput = {
+        metadata: {
+          description: 'cross-currency'
         },
-        5,
-        0.5
-      )
-
-      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
-        expect.objectContaining({
-          type: WebhookEventType.OutgoingPaymentCreated,
-          data: expect.any(Object)
-        })
-      )
-      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
-        expect.objectContaining({
-          type: WebhookEventType.OutgoingPaymentCompleted,
-          data: expect.any(Object)
-        })
-      )
-    })
-    test('Get Outgoing Payment', async (): Promise<void> => {
-      const payment = await c9.adminClient.getOutgoingPayment(
-        outgoingPayment.id
-      )
-      expect(payment.state).toBe(OutgoingPaymentState.Completed)
-      expect(payment.receiveAmount.value).toBe(amountValueToSend)
-      expect(payment.sentAmount.value).toBe(amountValueToSend)
+        incomingAmount: {
+          assetCode: 'EUR',
+          assetScale: 2,
+          value: value as unknown as bigint
+        },
+        walletAddressUrl:
+          'https://happy-life-bank-test-backend:4100/accounts/lars'
+      }
+
+      const receiver = await createReceiver(createReceiverInput)
+      assert(receiver.incomingAmount)
+      assert(
+        receiver.incomingAmount.assetScale === senderWalletAddress.assetScale
+      )
+      const receiverAssetCode = receiver.incomingAmount.assetCode
+
+      const quote = await createQuote(senderWalletAddressId, receiver)
+      const outgoingPayment = await createOutgoingPayment(
+        senderWalletAddressId,
+        quote
+      )
+      const payment = await getOutgoingPayment(outgoingPayment.id, value)
+      const exchangeRate =
+        c9.config.seed.rates[receiverAssetCode][senderAssetCode]
+      const { sentAmount, receiveAmount } = payment
+
+      if (exchangeRate > 1) {
+        expect(sentAmount.value).toBeGreaterThan(receiveAmount.value)
+      } else if (exchangeRate < 1) {
+        expect(sentAmount.value).toBeLessThan(receiveAmount.value)
```

Comment:
will this always be true though? depending on the fees, you might run into a scenario where the exchange rate is 0.98 or something, but with an added fee, this will fail this assertion. I guess what this means is that:

> I suppose maybe if really want to verify those amounts I can just hardcode what they currently are and state the assumptions in a comment (fee, rates, starting amount).

hardcoding the assumptions will always have to be the case.

With that said, I think we should either directly verify the amount, or not do the check at all. We can also leave it for a future enhancement, if you want. Thoughts?

Reply:
I thought of that so I changed the fee to 0, but I guess we would need to add some assertions checking the fee being 0. Which leads to you point about still having to hardcode assumptions.

I suppose making assertions (not jest `expect`, but `assert`) for fees, exchange rates, asset scale, asset code, and then expecting certain values is probably the better way to go. I'l include it here - this PR is still waiting on approval of the parent and the parents is waiting on the node 20 changes.

Reply:
add [0dda0e4](https://github.com/interledger/rafiki/pull/2608/commits/0dda0e41afc85503c9fc8b5a9864ad91f71141f4)

feel free to give anymore feedback on it but im merging this branch into the parent.